### PR TITLE
Update hardhat.py

### PR DESCRIPTION
**Pull request by alecmaly**
Patch to resolve issues with old compile versions.

Tested on: https://github.com/code-423n4/2023-04-ens

Tested w/ slither which uses crytic-compile as a dependency
```bash
slither-doctor .
slither .
```

### CODE REVIEW

File: crytic_compile/platform/hardhat.py [hunk 1]
```diff
@@ -132,7 +132,10 @@ def hardhat_like_parsing(
                         )
 
                     source_unit = compilation_unit.create_source_unit(path)
-                    source_unit.ast = info["ast"]
+                    try:
+                        source_unit.ast = info["ast"]
+                    except:
+                        source_unit.ast = info["legacyAST"]
```
Comment (0xalpharush) [inline crytic_compile/platform/hardhat.py:138]:
```suggestion
                    source_unit.ast = info.get("legacyAST", "ast")
```

Comment (alecmaly) [reply crytic_compile/platform/hardhat.py]: Good suggestion!

Comment (alecmaly) [reply crytic_compile/platform/hardhat.py]: Should we swap the "legacyAST" and "ast" so it attempts to look at the modern ast first?

Comment (0xalpharush) [reply crytic_compile/platform/hardhat.py]: Yeah, that makes sense

File: crytic_compile/platform/hardhat.py [hunk 2]
```diff
@@ -132,7 +132,7 @@ def hardhat_like_parsing(
                         )
 
                     source_unit = compilation_unit.create_source_unit(path)
-                    source_unit.ast = info["ast"]
+                    source_unit.ast = info.get("legacyAST", "ast")
```
Comment (0xalpharush) [inline crytic_compile/platform/hardhat.py:135]:
```suggestion
                    source_unit.ast = info.get("legacyAST", info["ast"])
```
Sorry realized my suggestion was incorrect

**Review by alecmaly — COMMENTED** [review]
My apologies, I got totally side tracked on this.

I believe this should be the code. Info.get() will return info["ast"] if it exists and info["legacyAST"] otherwise.

```
info.get("ast", "legacyAST")
```

I'm not at a computer to test right now but I believe this is the change that needs to happen.

**Review by alecmaly — COMMENTED** [review]
<phone posted the same comment twice and I cannot delete this? Please ignore.>

**Comment by 0xalpharush** [comment]
Thanks @alecmaly! This was merged as part of https://github.com/crytic/crytic-compile/pull/455

**Pull request closed without merging**